content/en/blog/2022/2022-03-15-k8s-triage-bot-helper-ci-job/K8s Triage bot Helper CI Job_ ‘make update’.md

```diff
@@ -0,0 +1,129 @@
+
+After you run these, you might see either of the below issues:
+
+---
+Using Codespaces:
+
+![codespaces error output](codespaces.png)
+![codespaces error output](codespaces2.png)
+
+---
+
+Using Visual Studio Code:
+
+![vscode error output](vscode.png)
+
+### Potential solution to the above error:
+1. Remove the Makefile and Makefile.generated_files files:
+
+```
+rm Makefile Makefile.generated_files
+```
+
+
+2. Create the symlinks:
+
+```
+ln -s build/root/Makefile Makefile
+```
+
+```
+ln -s build/root/Makefile.generated_files Makefile.generated_files
+```
+
+> PS: takes adequate time to generate the scripts on windows.
+
+## The current situation
+
+Many contributors do not have access to powerful environments in which to run `make update` or `make verify`. They can utilise *vscode/wsl/codespaces* and other tools to recommend modifications, but they might get tripped up by `make verify` scripts because in many cases, we end up with files that needs to be re-generated. It's a tall order for them to scan the build log from `make verify` to determine which specific scripts in `hack/` directory they need to run.
+
+### Solution
+
+Two solutions are proposed by [@dims](https://github.com/dims) for the time being:
+
+* **Short Term:** add a CI job that folks can trigger when they need it, that runs `make update` and provides a zip archive that they can be told to download. The archive includes all the files that changed as a result of running `make update`.
+*  **Long Term:** add a new Prow bot command that generates an additional commit on their PR.
+
+### Short Term Solution
+
+Once we realized the problem was that some of the verify scripts needed either linux to work better or a large CPU/Memory even with linux, we decided to add a new CI job. This CI job is named “pull-kubernetes-update”. You can trigger this CI job by running “/test pull-kubernetes-update” bot command in any PR. This CI job runs `make update` and generates a zip file named “updated-files.zip” in the artifacts directory for that job. Folks can then download the zip which has the changes made when `make update` was run and update their PR with the newly updated code.
+
+**Steps:**
+
+Make sure to rebase your 'working branch' to 'master' to align with current changes made to the repository. 
+
+On your PR, write a new comment that contains only the text: 
+
+```
+/test pull-kubernetes-update
+```
+
+![pull-kubernetes-update command](tpku_command.png)
+
+You will see the automated check lists and details.
+
+![command preview on github](checklists.png)
+
+Once the checks are complete, click the `Details` link on `pull-kubernetes-update` job to go to the artifacts directory and  download the `updated-files.zip` file.
+
+![automated checklist of tests appearing after update command execution](completed_checklist.png)
+
+![completed checklist of tests](artifacts.png)
+
+![artifacts directory](jenkins.png)
+
+Now, update the PR by adding the extracted files you downloaded. 
+
```

Comment:
There's no heading or steps about using the long term solution. Folks might wonder where that has gone.

My guess is that the long term solution is still a work in progress and you can't use it yet? Whatever the story is, let's make that really clear.

Reply:
I have an idea how to arrange this better, suggestion coming.